source/common/stats/symbol_table_impl.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <algorithm>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "envoy/common/exception.h"
+#include "envoy/stats/symbol_table.h"
+
+#include "common/common/assert.h"
+
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Stats {
+
+class SymbolTableImpl : public SymbolTable {
+public:
+  std::vector<Symbol> encode(const std::string& name) override {
+    std::vector<Symbol> symbol_vec;
+    std::vector<std::string> name_vec = absl::StrSplit(name, '.');
```

Comment:
Keep in mind that some prefixes/values contain `.` (ipv4 addresses, for instance), so this way of splitting won't necessarily line up with the individual elements that compose stats.  I don't think it'll be a problem, however, because it'll just mean that some symbols will be more subdivided than one might expect.

Reply:
This is a good note -- I'd bet that alongside other common patterns this implementation will be inefficient for large numbers of small string segments. We could do a first pass on encoding to regex-match IP addresses and encode them as unique symbols -- do you think this is worth doing in a separate PR?

Reply:
Yes, definitely a separate PR.  I wouldn't worry about it here.  Another option would be to change how IP addresses are represented in stats by changing their delimiter, but I think that's probably a discussion for a different PR :).

Reply:
Just leave a TODO for now I reckon.

Reply:
I think actually we don't want to encode each IP as a unique symbol -- especially for Envoys with large throughput the cost of encoding each IP as a unique token (mapping an int to... another int?) might be too expensive. I'll put a TODO here for myself to think about this in the future.

Reply:
Done.